app/models/post.rb

```diff
@@ -1,3 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  after_save :update_user_post_count
```

Comment:
- [**Kudos** ] Awesome job putting the `update_user_post_count` method as private and using the `after_sate` callback to automatically call it after the post object is saved to the database. :+1:

Reply:
@ridwanediallo hello, on my side it is working very well without giving that error
![pic](https://github.com/wolterbwambale/blog_app/assets/104980642/00c855a8-be9d-40f3-a37b-ddc246eec7d4)